lib/cocoapods-core/cdn_source.rb

```diff
@@ -0,0 +1,269 @@
+require 'cocoapods-core/source'
+require 'rest'
+require 'concurrent'
+
+module Pod
+  # Subclass of Pod::Source to provide support for CDN-based Specs repositories
+  #
+  class CDNSource < Source
+    # @param [String] repo The name of the repository
+    #
+    def initialize(repo)
+      # Optimization: we initialize startup_time when the source is first initialized
+      # and then test file modification dates against it. Any file that was touched
+      # after the source was initialized, is considered fresh enough.
+      @startup_time = Time.new
+
+      @executor = Concurrent::ThreadPoolExecutor.new(
+        :min_threads => 50,
```

Comment:
min_threads should probably be 1, from my experiences working on bundler, not every system is able to spin up 50 threads